src/Adapter/MSTestAdapter.PlatformServices/Execution/UnitTestRunner.cs

```diff
@@ -172,8 +172,11 @@ internal async Task<TestResult[]> RunSingleTestAsync(TestMethod testMethod, IDic
                 DebugEx.Assert(testMethodInfo is not null, "testMethodInfo should not be null.");
 
                 // Keep track of all non-runnable methods so that we can return the appropriate result at the end.
-                _assemblyFixtureTests.TryAdd(testMethod.AssemblyName, testMethodInfo.Parent.Parent);
-                _classFixtureTests.TryAdd(testMethod.AssemblyName + testMethod.FullClassName, testMethodInfo.Parent);
+                if (MSTestSettings.CurrentSettings.ConsiderFixturesAsSpecialTests)
```

Comment:
do we need a new test for this?

Reply:
There is no behavior change to test. The dictionary is only read when ConsiderFixturesAsSpecialTests is true. So we avoid adding to the dictionary (and allocating strings) that are never read.